monai/apps/auto3d/data_analyzer.py

```diff
@@ -0,0 +1,644 @@
+        Caller:
+            get_all_case_stats: The function iterates dataset.datalist and call get_case_stats to generate stats. Then
+                                get_case_summary is called to combine results and save_yaml is called to save results.
+
+    Custimize statistics calculation:
+        Write a new function for indivisual case:
+            _get_your_stats(self):
+                1. processed_data = retrive processed data from self.data, if not exist, process data (like cropping)
+                2. define a dict case_stats = {'stats1': calculate(processed_data)}. Notice the data may be a list of
+                data from different modalities. So calculate(processed_data) should return a list of stats. case_status
+                will be written to the yaml file
+                3. create a dict case_stats_summary = {'stats1': summary_function}. The summary_function will process
+                a list of ['stats1','stats1',...] from all cases in self.dataset.datalist.
+                4. update self.data with processed_data and update self.gather_summary to register case_stats_summary,
+                and return case_stats.
+        Add _get_your_stats to self._hardcode_functions.
+    """
+
+    def __init__(
+        self,
+        datalist: Union[str, Dict],
+        dataroot: str,
+        output_yaml: str = "./data_stats.yaml",
+        average: bool = False,
+        do_ccp: bool = True,
+        device: str = "cuda",
+        worker: int = 2,
+    ):
+        """
+        Initializer will load the data and register the functions for data statistics gathering.
+        """
+        self.output_yaml = output_yaml
+        files, _ = datafold_read(datalist=datalist, basedir=dataroot, fold=-1)
+        ds = data.Dataset(
+            data=files,
+            transform=transforms.Compose(
+                [
+                    transforms.LoadImaged(keys=["image", "label"]),
+                    transforms.EnsureChannelFirstd(keys=["image", "label"]),  # this creates label to be (1,H,W,D)
+                    transforms.Orientationd(keys=["image", "label"], axcodes="RAS"),
+                    transforms.EnsureTyped(keys=["image", "label"], data_type="tensor"),
+                    # some dataset has onehot label size of (H,W,D,C). Only allow label index of size (1,H,W,D)
+                    transforms.Lambdad(
+                        keys="label", func=lambda x: torch.argmax(x, dim=0, keepdim=True) if x.shape[0] > 1 else x
+                    ),
+                    transforms.SqueezeDimd(keys=["label"], dim=0),  # make label (H,W,D)
+                ]
+            ),
+        )
+        self.dataset = data.DataLoader(ds, batch_size=1, shuffle=False, num_workers=worker, collate_fn=lambda x: x)
+        # Whether to average all the modalities in the summary
+        # If SUMMERY_AVERAGE is set to false,
+        # the stats for each modality are calculated separately
+        self.SUMMERY_AVERAGE = average
```

Comment:
Slight change to the variable name and references to it in comments: "SUMMERY_AVERAGE" -> "SUMMARY_AVERAGE"

Reply:
Thanks for catching that! Fixed that and also a few documentation typo in 61ec647003febaaec9c7937e47390a865b2bb003